generated/java/java_batch.py

```diff
@@ -0,0 +1,422 @@
+import os
+import sys
+import json
+import fnmatch
+import argparse
+import subprocess
+from artman.config import loader
+from artman.config.proto.config_pb2 import Artifact, Config
+from google.protobuf import json_format
+import yaml
+import shutil
+from tempfile import mkstemp
+from os import fdopen, remove
+
+# list of APIs that will be generated
+APIS = [
+    # shared package
+    'core',
+    'appengine',
+    'iam',
+    # gapic API
+    'bigquerydatatransfer',
+    'bigtable',
+    'bigtableadmin',
+    'container',
+    'dataproc_v1',
+    'datastore',
+    'dialogflow_v2beta1_java',
+    'dlp_v2beta1',
+    'dlp_v2beta2',
+    'errorreporting',
+    'firestore',
+    'language_v1',
+    'language_v1beta2',
+    'logging',
+    'longrunning',
+    'monitoring',
+    'pubsub',
+    'oslogin_v1',
+    'spanner',
+    'spanner_admin_database',
+    'spanner_admin_instance',
+    'speech_v1',
+    'speech_v1beta1',
+    'speech_v1p1beta1',
+    'cloudtrace_v1',
+    'cloudtrace_v2',
+    'videointelligence_v1beta1',
+    'videointelligence_v1beta2',
+    'videointelligence_v1',
+    'vision_v1',
+    'vision_v1p1beta1',
+]
+
+# list of APIs which don't generate proto library
+PROTO_EXCLUSION = ['longrunning']
+
+# list of APIs which don't generate grpc library
+GRPC_EXCLUSION = ['appengine', 'longrunning']
+
+# list of APIs which `copy` subcommand doesn't copy from staging to gcj repo
+COPY_EXCLUSION = [
+        'speech_v1p1beta1',
+        'dlp_v2beta2',
+        'datastore',
+        'spanner_admin_database',
+        'spanner_admin_instance',
+        'spanner',
+]
```

Comment:
I would actually prefer the default to be no-copy, and then flip to copy. The reason is that when a new API is configured, there can be a period where we don't want to put the generated code into google-cloud-java yet; this scenario requires two flips (1. add to the exclusion list, 2. remove off the exclusion list). One way to make this easier to manage is to make the API list have attributes for whether the copy is on or not, so that we don't need two separate lists for proto/grpc generation vs copying.